src/main/java/org/wso2/carbon/connector/amazonsqs/utils/Utils.java

```diff
@@ -0,0 +1,283 @@
+import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeNameForSends;
+import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeValue;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SqsResponseMetadata;
+
+import javax.xml.stream.XMLStreamException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class contain required util methods for to Amazon sqs connector.
+ */
+public class Utils {
+
+    private static final Log log = LogFactory.getLog(Utils.class);
+
+    /**
+     * Sets the error code and error detail in message
+     *
+     * @param messageContext Message Context
+     * @param error          Error to be set
+     */
+    public static void setErrorPropertiesToMessage(MessageContext messageContext, Error error, String errorDetail) {
+        messageContext.setProperty(Constants.PROPERTY_ERROR_CODE, error.getErrorCode());
+        messageContext.setProperty(Constants.PROPERTY_ERROR_MESSAGE, error.getErrorMessage());
+        messageContext.setProperty(Constants.PROPERTY_ERROR_MESSAGE, errorDetail);
+    }
+
+    public static OMElement createOMElement(String elementName, Object value) {
+        OMElement resultElement = null;
+        try {
+            if (value != null) {
+                resultElement = AXIOMUtil.
+                        stringToOM("<" + elementName + ">" + value
+                                + "</" + elementName + ">");
+            } else {
+                resultElement = AXIOMUtil.
+                        stringToOM("<" + elementName
+                                + "></" + elementName + ">");
+            }
+        } catch (XMLStreamException | OMException e) {
+            log.error("Error while generating OMElement from element name" + elementName, e);
+        }
+        return resultElement;
+    }
+
+    public static String createUrl(MessageContext messageContext, SqsConnection sqsConnection)
+            throws MalformedURLException, SqsInvalidConfigurationException {
+        try {
+            String queueName = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_NAME);
+            String queueId = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_ID);
+            String queueUrl = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_URL);
+            if (StringUtils.isEmpty(queueName) && (StringUtils.isEmpty(queueId) || StringUtils.isEmpty(queueUrl))) {
+                throw new SqsInvalidConfigurationException("Missing queue info: Should provide value of queue url or " +
+                        "queue name and queue id.");
+            }
+            if (StringUtils.isNotEmpty(queueUrl)) {
```

Comment:
Shall we move this above and refactor this method. Also better to use `StringUtils.isNotBlank` to handle whitespaces as well. Please update the other references as well.